scom/stackstate_checks/scom/scom.py

```diff
@@ -82,114 +52,136 @@ def sendAlerts(self, session, scom_ip, domain, username, password, component_id)
                     "time_added:%s" % event.get("timeadded")
                 ]
             })
+    def get_health_state(self,healthIconUrl):
+        if re.search('StatusOkComplete', healthIconUrl, flags=re.IGNORECASE) or re.search('healthy_ex', healthIconUrl,flags=re.IGNORECASE):
+           return 'Healthy'
+        elif re.search('StatusWarning', healthIconUrl, flags=re.IGNORECASE) or re.search('warning_ex', healthIconUrl,flags=re.IGNORECASE):
+           return 'Deviating'
+        elif re.search('StatusError', healthIconUrl, flags=re.IGNORECASE) or re.search('critical_ex', healthIconUrl,flags=re.IGNORECASE):
+           return 'Error'
+        elif re.search('StatusUnknownGreen', healthIconUrl, flags=re.IGNORECASE) or re.search('maintenance_ex',healthIconUrl,flags=re.IGNORECASE):
+           return 'Not Monitored'
+        else:
+           return str(healthIconUrl.split("/")[-1])
+
+    def get_type(self,session,component_id,domain, username, password, scom_ip):
+        if self.requests_counter > self.requests_threshold:
+           return
+        type_data = {
+                                "criteria": "(Id = '"+component_id+"')"
+                                }
+        type_response = (session.post('http://'+ scom_ip +'/OperationsManager/data/scomObjects',auth=HttpNtlmAuth(domain + '\\' + username, password),json=type_data["criteria"])).json()
```

Comment:
I think it's unnecessary to create the `type_data` dictionary if you are only using it for `type_data["criteria"]`

Reply:
fixed, i got rid of type_data